processing/src/main/java/org/apache/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -254,10 +256,10 @@ public <T extends LogicalSegment> List<T> filterSegments(SegmentMetadataQuery qu
 
   @VisibleForTesting
   public static SegmentAnalysis mergeAnalyses(
-      @Nullable String dataSource,
+      Set<String> dataSources,
```

Comment:
why use all datasources now, instead of first as before? I believe that the SegmentAnalysis is for single table?

Reply:
Oh I see, its equivalent to before since the first datasource is checked first, and now we have added protection that if first segment didn't belong to first dataSource in list but to a subsequent one, then merge analysis will still work.

Reply:
yeah, this is for `UnionDataSource` - previously, the caller was passing only the first one [here](https://github.com/apache/druid/pull/14560/files#diff-acc53e0e891879963c30d5c8b349347f6b7d96c0560992a3c8fe1f9f3d2ddd3bL142). Now we pass in all the datasources and determine the datasource in this function